include/zephyr/net/net_if.h

```diff
@@ -2526,6 +2526,23 @@ int net_if_resume(struct net_if *iface);
 bool net_if_is_suspended(struct net_if *iface);
 #endif /* CONFIG_NET_POWER_MANAGEMENT */
 
+/**
+ * @brief Check if the network interface supports Wi-Fi.
+ *
+ * @param iface Pointer to network interface
+ *
+ * @return True if interface supports Wi-Fi, False otherwise.
+ */
+bool net_if_is_wifi(struct net_if *iface);
+
+/**
+ * @brief Get first Wi-Fi network interface.
+ *
+ * @return Pointer to network interface, NULL if not found.
+ */
+struct net_if *net_if_get_first_wifi(void);
```

Comment:
question, why the first wifi IF? If the platform has <n> wifi interfaces how would one specify an arbitrary IF?

Reply:
This is just a commonly used helper, one can do `net_if_foreach` + `net_if_is_wifi` and choose any Wi-Fi interface of interest based on any criterion. If its needed we can even add `net_if_wifi_foreach` in the future as a shortcut.